### ISSUE TITLE
Error when generating migration

### ISSUE THREAD

**Issue by chris-doco** [GetDutchie/brick#352]
Flutter version: `3.10.5`

Pubspec versions:
```
brick_offline_first: ^3.0.3
brick_offline_first_with_rest: ^3.0.1
brick_offline_first_with_rest_build: ^3.1.0
```

In a project I'm working on we started using Brick before version 1, but since upgrading it to Brick 2 and even when we upgraded to Brick 3 we keep getting this error when we run `flutter pub run build_runner build --delete-conflicting-outputs --verbose`.

This project is a fairly large project with 84 migration files (which were generated in Brick version 1 or before).

```
[INFO] Entrypoint:Generating build script...
[INFO] Entrypoint:Generating build script completed, took 156ms

[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Reading cached asset graph...
[INFO] BuildDefinition:Reading cached asset graph completed, took 55ms

[INFO] BuildDefinition:Checking for updates since last build...
[INFO] BuildDefinition:Checking for updates since last build completed, took 819ms

[INFO] Build:Running build...
[INFO] Heartbeat:1.1s elapsed, 84/91 actions completed.
[INFO] Heartbeat:2.2s elapsed, 84/91 actions completed.
[INFO] Heartbeat:9.6s elapsed, 84/91 actions completed.
[INFO] Brick:Aggregating models and migrations...
[SEVERE] brick_offline_first_with_rest_build:brick_schema_builder on lib/models_and_migrations.brick_aggregate.dart:

Unsupported operation: Null
package:source_gen/src/constants/reader.dart 123:33                                   _NullConstant.objectValue
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 76:64      MigrationGenerator._migrationCommandsFromReader.<fn>
dart:_internal                                                                        ListIterable.toList
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 128:8      MigrationGenerator._migrationCommandsFromReader
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 42:13      MigrationGenerator.expandAllMigrations.<fn>
dart:core                                                                             Iterable.toList
package:brick_sqlite/src/db/migration_manager.dart 54:58                              MigrationManager.latestMigrationVersion
package:brick_sqlite_generators/src/sqlite_schema/sqlite_schema_generator.dart 82:34  SqliteSchemaGenerator._createNewSchema
package:brick_sqlite_generators/src/sqlite_schema/sqlite_schema_generator.dart 24:23  SqliteSchemaGenerator.generate
package:brick_sqlite_generators/src/builders/sqlite_schema_builder.dart 19:36         SchemaBuilder.build

[SEVERE] brick_offline_first_with_rest_build:brick_new_migration_builder on lib/models_and_migrations.brick_aggregate.dart:

Unsupported operation: Null
package:source_gen/src/constants/reader.dart 123:33                                   _NullConstant.objectValue
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 76:64      MigrationGenerator._migrationCommandsFromReader.<fn>
dart:_internal                                                                        ListIterable.toList
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 128:8      MigrationGenerator._migrationCommandsFromReader
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 42:13      MigrationGenerator.expandAllMigrations.<fn>
dart:core                                                                             Iterable.toSet
package:brick_sqlite_generators/src/sqlite_schema/migration_generator.dart 135:59     MigrationGenerator.generate
package:brick_sqlite_generators/src/sqlite_schema/sqlite_schema_generator.dart 49:31  SqliteSchemaGenerator.createMigration
package:brick_sqlite_generators/src/builders/new_migration_builder.dart 20:36         NewMigrationBuilder.build

[INFO] Brick:Generated brick.g.dart, took 6ms
[INFO] Build:Running build completed, took 10.6s

[INFO] Build:Caching finalized dependency graph...
[INFO] Build:Caching finalized dependency graph completed, took 38ms

[SEVERE] Build:
Failed after 10.7s
```

It seems to happen when it's generating be caused by some Null value from `migration_generator.dart`.

Thank you.

**Comment by tshedor** [comment]
Hey @chris-doco thanks for using Brick. First, this is awesome:

> This project is a fairly large project with 84 migration files (which were generated in Brick version 1 or before).

I've never gotten to that scale before. Hat tip to you and your (potential) team.

Can you tell me what version of Dart and Flutter you're using and what version (from `pubspec.lock`) of `brick_sqlite_generators` you're using? 

It looks like you've got a migration that is not declaring the column type on an `InsertColumn` (i.e. `InsertColumn('name', Column.integer)`). That could be in your `schema.g.dart` or in a `.migration.dart` file. Does this sound familiar?

As an aside, you can consolidate old migrations together if you're certain there are no more users on that version. For example, if version 1 of your project used migrations 1,2,3,4,5 and version 1.2 used migrations 6,7,8 and everyone is on 1.1, you can combine migrations 1-5 into migration 1. Completely not necessary and not totally for the faint of heart, but it can help declutter your codebase in a pinch. If you do it right, re-running `build runner` will not generate a new migration.

**Comment by tshedor** [comment]
@chris-doco it looks like you're on a different daytime than I am; the other way you can debug is opening `.dart_tool/build` (`.dart_tool` will be in the root of your project). Copy/paste the contents of `models_and_migrations.brick_aggregate.dart` into a file in your `lib` and see if the analyzer complains about any `InsertColumn`. My uneducated guess is it's related to the change in #44 that changed `int` as an acceptable parameter to `Column`. You might have an `int` in there that should be a `Column.integer`.

**Comment by chris-doco** [comment]
Hi, thanks for the quick response.

I have found the issue from your suggestion of checking the `models_and_migrations.brick_aggregate.dart` file!

It had an import for `package:flutter/material.dart` which had a `Column` class that conflicted with `enum Column` from `package:brick_sqlite/db.dart`.
This import was due to one of the models having that import. I have modified the model to not use the `package:flutter/material.dart` import and it now generates migrations properly.

Thank you!
